frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsAndSoil.tsx

```diff
@@ -50,88 +58,132 @@ const FieldsAndSoilComponent: React.FC<InputModuleProps> = ({ farmDetails, updat
   ): void => {
     setTimeout(() => {
       const farmInformation: FarmDetailsInterface = { ...farmDetails };
-      farmInformation.FieldName = values.FieldName;
-      farmInformation.Area = values.Area;
-      farmInformation.Comments = values.Comments;
+      farmInformation.Fields[fieldIndex].FieldName = values.FieldName;
+      farmInformation.Fields[fieldIndex].Area = values.Area;
+      farmInformation.Fields[fieldIndex].Comments = values.Comments;
       updateFarmDetails(farmInformation);
       setSubmitting(false);
     }, 400);
   };
 
   const addFarmInfo = (values: SubmissionValues) => {
     const farmInfo: FarmDetailsInterface = { ...fieldsInfo };
-    farmInfo.FieldName = values.FieldName;
-    farmInfo.Area = values.Area;
-    farmInfo.Comments = values.Comments;
+    farmInfo.Fields.push({
+      FieldName: values.FieldName,
+      Area: values.Area,
+      Comments: values.Comments,
+    });
+    console.log(farmInfo);
     setFieldsInfo(farmInfo);
+    setFieldIndex((prevIndex) => prevIndex + 1);
+    setSubmitted(true);
   };
 
+  const addNewField = () => {
+    setFieldAdd(true);
+  };
+  console.log(`Field add state is ${fieldAdd}`);
+  console.log(`Field submitted state is ${isSubmitted}`);
```

Comment:
Leftover logs? or development logs?

Reply:
Dev logs, I was checking if my state hooks was working as intended. Showing form at first then disappearing after add button triggers a setState then lists appears, add new fields triggers another state while form appears. It's a bit of a frustration at first.

Reply:
Will delete it as I clean up my code.